### DISCUSSION TITLE
Referring to Chapter By Number

### DISCUSSION THREAD

**Discussion by iwelch** [jgm/pandoc#10235] [Q&A]
I use markdown for writing books now.  The main output targets are latex's book and the epub format, as well as html (which numbers chapters only by virtue of a style file).  Because the writing in some places needs to refer to earlier chapters ("In Chapter X, ..."), I need to refer back to number X, aka \label{} and \ref{} in LaTeX.  I know that the pandoc-xnos tools provide this kind of facility.  Is this the recommended solution now, or is there some mechanism built into base pandoc?

**Comment by jgm** [comment]
No, there's nothing built in that does this.  (For LaTeX you can just use `\ref`, but that won't be portable to EPUB.)

It would be possible to create a filter that handles this, but if pandoc-xnos does it, why reinvent the wheel.

**Comment by iwelch** [comment]
dankeschoen.

**Comment by tarleb** [comment]
I've been hacking away at [querverweis](https://github.com/tarleb/querverweis), as I needed a filter that doesn't require python. Another motivation was that pandoc-xnos appears to be unmaintained right now.
